libraries/AP_HAL_Linux/Led_Sysfs.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+   Copyright (C) 2017 Mathieu Othacehe. All rights reserved.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Led_Sysfs.h"
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include <unistd.h>
+
+#include <AP_HAL/AP_HAL.h>
+
+static const AP_HAL::HAL &hal = AP_HAL::get_HAL();
+
+namespace Linux {
+
+Led_Sysfs::Led_Sysfs(const char *led_name)
+    : _led_name(led_name)
+{
+}
+
+Led_Sysfs::~Led_Sysfs()
+{
+    if (_brightness_fd)
+        close(_brightness_fd);
```

Comment:
I personnaly prefer to always use parenthesis, it prevents future and indentation problem.